source/App.cpp

```diff
@@ -424,7 +438,7 @@ void App::updateParameters(int frameDelay, float frameRate) {
 	float dt = 0;
 	if (frameRate > 0) dt = 1.0f / frameRate;
 	else dt = 1.0f / float(window()->settings().refreshRate);
-	setFrameDuration(dt, GApp::REAL_TIME);
+	setFrameDuration(dt, m_simTimeStep);
```

Comment:
This is another case of circular logic in calling  `GApp::setFrameDuration()`, which is the setter for `m_simTimeStep`. We want to set the frame duration using `REAL_TIME` here based on the parameters passed into the function. I can't think of a case where we would use anything of than a `REAL_TIME` so I think it makes sense to hard code that here or offer it up as an additional (optional) function parameter.

Unless there is a reason I'm missing I think inheriting the time-base here and the time somewhere else to make 2 calls to `GApp::setFrameDuration()` is more confusing than necessary.

Reply:
I'd argue it's better to set a value only when it needs to change rather than hammering it every frame. The reason here is the test wants to set GApp::MATCH_REAL_TIME_TARGET for a fixed timestep and needs this not to change every time it calls oneFrame().